### ISSUE TITLE
Add pipeline check for inferred equivalent classes within an ontology

### ISSUE THREAD

**Issue by cmungall** [obophenotype/upheno#149]
See for example
https://github.com/obophenotype/human-phenotype-ontology/issues/522

Oort should already produce this report. Investigate options for making this more promiment

**Comment by cmungall** [comment]
PR for HP: https://github.com/obophenotype/human-phenotype-ontology/pull/1031

@nicolevasilevsky can you start working on these:

https://github.com/obophenotype/human-phenotype-ontology/blob/9c36be164c2a85c09b492633b3b32bf0169448a7/src/ontology/reports/equivalentClasses.csv

**Comment by nicolevasilevsky** [comment]
what is the action item here? Is it to determine why these classes are considered equiv? For most of these, I think the logical definitions are the same, which is why they are being inferred as equiv classes.

**Comment by cmungall** [comment]
Yes, immediate action is to fix the logical defs. Remember, a bad axiom is often worse than no axiom.

**Comment by drseb** [comment]
Please do not just remove axioms. Checking the hierarchy and/or class merging also has to be considered

**Comment by cmungall** [comment]
good point. Many of these are 'bad smells' indicating a need to merge.

@drseb, how would you like to proceed? Should Nicole create a new ticket 
for each one of these, where we can give the +1/-1 on ways to proceed? 
That's how I would do it, but I'm a splitter when it comes to tickets, 
you may prefer to group.

**Comment by drseb** [comment]
Ok. This is probably a good start:

- If the two classes are "highly similar" (read text-def) and one of them just needs another qualifier like progressive, localised, pathologic,... just add the qualifier. If the qualifier is not in PATO, but is a candidate for PATO, then let's discuss.
- If the two terms are highly similar, but one of them has a text-def that clearly separates him from the other term, and we don't have way to express this difference in OWL, remove the logical def of the more specific class. (Check set of ancestor classes)
- If the two classes are really equivalent, check the set of ancestors. If one of them has ancestors that should apply to the other class, then file an HPO-ticket.
- If the two classes are really equivalent, check the set of ancestors. If one of them INCORRECTLY has ancestors that the other does not have, then file an HPO-ticket.
- If the two classes are really equivalent (i.e. also have the 'same' set of ancestors) then merge them. (Please be very cautious with term merging as this has consequences for annotation data all over the world. Ask Peter, Chris, Melissa, or me for merge-instructions if you need a reminder)

Please handle them one-by-one and be verbose in commits. Maybe do only 5-10 per day such that I have a chance to check them (because I still have only limited time per day to work at the moment)

Let me know if I forgot something.

@cmungall @pnrobinson @mellybelly

**Comment by cmungall** [comment]
I am in perfect agreement. @nicolevasilevsky will you make a start?

**Comment by nicolevasilevsky** [comment]
I will! :)

**Comment by cmungall** [comment]
Robot should be used to report inferred equivalencies moving forward. Asking @dougl1sqrd if you have questions.

**Comment by nicolevasilevsky** [comment]
No, a lot of these tickets are still open. I'll look at these open tickets and see if I can move them along.
